QUICKSTART.md

```diff
@@ -0,0 +1,206 @@
+
+```bash
+export PATH="$HOME/.local/bin:$PATH"
+```
+
+Congratulations! You've successfully installed and set up hypha. Go to <TBD> to start your first training.
+
+---
+
+### Manual Installation (build from source)
+
+If you're on a different system or want to manually build hypha for other reasons, you can also build your project from source.
+
+```bash
+git clone https://github.com/hypha-space/hypha.git
+cd hypha
+```
+
+Install dependencies:
+
+```bash
+cargo build -r
+```
+
+You'll find your binaries in the projects target/release folder. Remember to move them to somewhere in your `$PATH`.
+
+---
+
+## Configuration
+
+All hypha components require some level of setup, also we need to prepare certificates.
+
+---
+
+### Certificates
+
+First, we need to create the necessary certificates, you can have a look at `hypha-certutil --help` for details
+
+For this quickstart guide, we're assuming the following commands have been executed:
+
+```bash
+hypha-certutil root --organization root && \
+hypha-certutil org --root-cert ./root-ca-cert.pem --root-key ./root-ca-key.pem -o quickstart && \
+hypha-certutil node --ca-cert ./quickstart-ca-cert.pem --ca-key ./quickstart-ca-key.pem -n gateway && \
+hypha-certutil node --ca-cert ./quickstart-ca-cert.pem --ca-key ./quickstart-ca-key.pem -n scheduler && \
+hypha-certutil node --ca-cert ./quickstart-ca-cert.pem --ca-key ./quickstart-ca-key.pem -n worker1 && \
+hypha-certutil node --ca-cert ./quickstart-ca-cert.pem --ca-key ./quickstart-ca-key.pem -n worker2 && \
+hypha-certutil node --ca-cert ./quickstart-ca-cert.pem --ca-key ./quickstart-ca-key.pem -n worker3 && \
+hypha-certutil node --ca-cert ./quickstart-ca-cert.pem --ca-key ./quickstart-ca-key.pem -n data1
+```
+
+---
+
+### Setting up nodes
+
+Next you need to set up your nodes, we're assuming the following configs for this getting started guide:
+
+---
+
+#### Gateway node
+
+You can generate a config for local development using `hypha-gateway init`:
+
+```bash
+hypha-gateway init -n gateway -o gateway-config.toml --exclude-cidr 192.0.2.0/24
+```
+
+Note: the `-n` / `--name` parameter controls the base name of the generated config files; for example `-n gateway` will produce `gateway-config.toml` (and related files) in the current directory.
+
+#### Scheduler node
+
+You can generate the necessary config using `hypha-scheduler init`:
+
+```bash
+hypha-scheduler init -n scheduler -o scheduler-config.toml --exclude-cidr 192.0.2.0/24 --gateway /ip4/127.0.0.1/tcp/8080
+```
+
+#### Worker nodes
+
+You can generate worker configs using `hypha-worker init`:
+
+```bash
+hypha-worker init -n worker1 -o worker1-config.toml --exclude-cidr 192.0.2.0/24 --gateway /ip4/127.0.0.1/tcp/8080 && \
+hypha-worker init -n worker2 -o worker2-config.toml --exclude-cidr 192.0.2.0/24 --gateway /ip4/127.0.0.1/tcp/8080 && \
+hypha-worker init -n worker3 -o worker3-config.toml --exclude-cidr 192.0.2.0/24 --gateway /ip4/127.0.0.1/tcp/8080
```

Comment:
The default configuration for workers doesn't include an `executor` configuration (changed with #112). So this configuration won't run anything.

Reply:
Workers need this in their configuration:

```toml
[[executors]]
class = "train"
name = "diloco-transformer"
runtime = "process"
cmd = "uv"
args = [
    "run",
    "--python", "3.12",
    "--no-project",
    "--with", "https://github.com/hypha-space/hypha/releases/download/v0.1.0/hypha_accelerate_executor-0.1.0-py3-none-any.whl",
    "--",
    "accelerate",
    "launch",
    "--config_file", "/etc/hypha/accelerate.yaml",
    "-m", "hypha.accelerate_executor.training",
    "--socket", "{SOCKET_PATH}",
    "--work-dir", "{WORK_DIR}",
    "--job", "{JOB_JSON}",
]
```

and parameter server needs this:

```toml
[[executors]]
class = "aggregate"
name = "parameter-server"
runtime = "parameter-server"
```

Reply:
nope, you can have both in your config (we tested/fixed). things work as they are now.